diff --git a/arch/arm/plat-aspeed/dev-i2c.c b/arch/arm/plat-aspeed/dev-i2c.c
old mode 100644
new mode 100755
index 5dd2af0..2585a6c
--- a/arch/arm/plat-aspeed/dev-i2c.c
+++ b/arch/arm/plat-aspeed/dev-i2c.c
@@ -1157,6 +1157,21 @@ static struct i2c_board_info __initdata mavericks_i2c_bus10[] = {
   {
     I2C_BOARD_INFO("fancpld", 0x66),
   },
+  {
+    I2C_BOARD_INFO("tmp75", 0x48),
+  },
+  {
+    I2C_BOARD_INFO("tmp75", 0x49),
+  },
+  {
+    I2C_BOARD_INFO("tmp75", 0x4a),
+  },
+  {
+    I2C_BOARD_INFO("tmp75", 0x4b),
+  },
+  {
+    I2C_BOARD_INFO("max6658", 0x4c),
+  },
 };
 
 /* i2c bus 11,12 on mavericks are not connected as i2c bus */
